x/epochs/abci.go

```diff
@@ -29,13 +30,16 @@ func BeginBlocker(ctx sdk.Context, k keeper.Keeper) {
 			epochStart = true
 			epochInfo.CurrentEpoch = epochInfo.CurrentEpoch + 1
 		}
-
 		if epochStart {
```

Comment:
This looks to me like its never getting ran after the first epoch? `epochStart` only ever get set to true if `epochInfo.EpochCountingStarted() == false`. But when thats true (for active epochs), this logic should never be getting ran?

Reply:
Oh wait no, theres a tabbing / spaces issue on line 28/29. (Can golint be ran)

Ok this is still being based on `CurrentEpochEnded` -- which I don't think is right. We wanted the end of the prior epoch to get detected within begin block, and then have the prior epoch ended here, and the next one began here -- all in one block.